class/practice3.2/src/Triangle.cpp

```diff
@@ -0,0 +1,24 @@
+#include "Triangle.hpp"
+#include <cmath>
+#include <string>
+#include <format>
+
+Triangle::Triangle(const double& first, const double& second, const double& third): a(first), b(second), c(third){}
+
+std::string Triangle::getArea(){
+    if (isValid()){
```

Comment:
let's rewrite with optional and move handling to a higher level of abstraction outside the class

```
std::optional<double> Triangle::calculateArea(){
    if (!isValid()) {
         return {};
    }
    double s = (a + b + c) / 2.0;
    double area = sqrt(s * (s - a) * (s - b) * (s - c));
    return area;
}
```